extensions/ql-vscode/src/ide-server.ts

```diff
@@ -11,11 +11,14 @@ import { ideServerLogger } from './logging';
 /** Starts a new CodeQL language server process, sending progress messages to the status bar. */
 export async function spawnIdeServer(config: QueryServerConfig): Promise<StreamInfo> {
   return window.withProgress({ title: 'CodeQL language server', location: ProgressLocation.Window }, async (progressReporter, _) => {
+    const args = process.env.DEBUG_LANGUAGE_SERVER
+      ? ['-J=-agentlib:jdwp=transport=dt_socket,address=localhost:9009,server=y,suspend=n,quiet=y', '--check-errors', 'ON_CHANGE']
```

Comment:
Nit: you could split this so it conditionally adds the debug flags to `args`, and then always adds the `check-errors ON_CHANGE`.